### PR TITLE
Analyzer.load helper, with more analysis smarts.

### PR DESCRIPTION
**Pull request by nevir**
* JSDoc tags are now being merged into element descriptors (when requested).
* Elements/properties are tagged w/ inferred metadata (private, configuration, etc).

This lets us move a lot of logic out of the doc viewer, and share it w/ other viewers.
See also: https://github.com/PolymerElements/iron-doc-viewer/pull/3/files

### CODE REVIEW

**Comment by nevir** [comment]
cc @mbleigh

**Comment by mbleigh** [comment]
:+1:

File: lib/analyzer.js [hunk 1]
```diff
@@ -96,6 +100,58 @@ var Analyzer = function Analyzer(attachAST,
...
+ * `href`.
+ *
+ * In order to properly filter paths, `href` _must_ be an absolute URI.
+ *
+ * @param {string} href The root import to begin loading from.
+ * @param {LoadOptions=} options Any additional options for the load.
+ * @return {Promise<Analyzer>} A promise that will resolve once `href` and its
+ *     dependencies have been loaded and analyzed.
+ */
+Analyzer.load = function load(href, options) {
```
Comment (garlicnation) [inline lib/analyzer.js]: Can you rename this? load() on the prototype and load() on the class seems ripe for confusion.

analyze()?

**Comment by garlicnation** [comment]
LGTM + nit.

File: lib/analyzer.js (hunk 1 above)
Comment (nevir) [inline lib/analyzer.js]: `analyze()` sounds good to me!

**Comment by azakus** [comment]
I don't see the source changes, only the build file changers.

**Pull request merged**